### ISSUE TITLE
mobilenet performs worse than inceptionv1 on size and speed

### ISSUE THREAD

**Issue by tbornt** [Zehaos/MobileNet#1]
I trained a mobilenet model using tensorflow flowers dataset. I'm not sure why the model size is larger than InceptionV1 and it costs more time on inference compared to inceptionV1.

**Comment by Zehaos** [comment]
A known bug: 
slim.separable_convolution2d used mistakenly

I am working on this.

**Comment by Zehaos** [comment]
I have used tf.nn.depthwise_conv2d instead. The model size should be right now.

**Comment by tbornt** [comment]
I set `num_outputs=None` in the `separable_convolution2d`. Does it work the same way as `depthwise_conv2d`?

**Comment by Zehaos** [comment]
Look good! Thanks. 
I will train the first model this weekend.

**Comment by kwotsin** [comment]
@Zehaos Would it be the same if you replaced your function for depthwise separable conv2d with the official function from tensorflow in https://github.com/tensorflow/tensorflow/blob/08cb0ba5431ef2508c6ca63c498023a84f7b4a18/tensorflow/python/layers/convolutional.py#L870 ?

It includes the depth_multiplier in the function as well, which could make the code more concise. Did you see any difference from implementing this official function instead?

**Comment by Zehaos** [comment]
@kwotsin According to tf doc,I think 'tf.nn.separable_conv2d()' combine depthwise conv and pointwise conv together, whithout bn nor relu between them, that is the problem.

**Comment by kwotsin** [comment]
@Zehaos Yes, but I think you can use `slim.batch_norm` as the `normalizer_fn` in `slim.conv2d` or `slim.separable_convolution2d` to use the batch norm. Relu is included by default in those 2 functions, and go by conv-->batch_norm-->relu.

**Comment by Zehaos** [comment]
@kwotsin yes, you are right. tf is so fexiable.
